cassovary-core/src/main/scala/com/twitter/cassovary/graph/MemoryMappedDirectedGraph.scala

```diff
@@ -51,48 +70,63 @@ class MemoryMappedDirectedGraph(file: File) extends DirectedGraph[Node] {
   private class MemoryMappedDirectedNode(override val id: Int) extends Node {
     val nodeOutboundOffset = outboundOffset(id)
     val nodeInboundOffset = inboundOffset(id)
-    override def outboundNodes(): Seq[Int] = new IndexedSeq[Int] {
+    def outboundNodes(): Seq[Int] = new IndexedSeq[Int] {
       val length: Int = outDegree(id)
       def apply(i: Int): Int =  data.getInt(nodeOutboundOffset + 4L * i)
-      override def toString(): String = "Memory Mapped IndexedSeq " + mkString("[", ",", "]")
     }
-    override def inboundNodes(): Seq[Int] = new IndexedSeq[Int] {
+    def inboundNodes(): Seq[Int] = new IndexedSeq[Int] {
       val length: Int = inDegree(id)
       def apply(i: Int): Int =  data.getInt(nodeInboundOffset + 4L * i)
-      override def toString(): String = "Memory Mapped IndexedSeq " + mkString("[", ",", "]")
     }
   }
 
-  override def getNodeById(id: Int): Option[Node] =
+  def getNodeById(id: Int): Option[Node] =
     if(0 <= id && id < nodeCount) {
       Some(new MemoryMappedDirectedNode(id))
     } else {
       None
     }
 
-  override def iterator: Iterator[Node] = (0 to nodeCount).iterator flatMap (i => getNodeById(i))
+  def iterator: Iterator[Node] = (0 to nodeCount).iterator flatMap (i => getNodeById(i))
 
-  override def edgeCount: Long = (outboundOffset(nodeCount) - outboundOffset(0)) / 4
+  lazy val edgeCount: Long = outboundOffset(nodeCount) - outboundOffset(0)
 
-  override lazy val maxNodeId: Int = nodeCount - 1
-  // Uncomment in future if maxNodeId becomes a method:
-  // override def maxNodeId = nodeCount - 1
+  override lazy val maxNodeId = nodeCount - 1
 
-  override val storedGraphDir = StoredGraphDir.BothInOut
+  val storedGraphDir = StoredGraphDir.BothInOut
 
-  def loadGraphToRam(): Unit = data.loadFileToRam()
+  /** Loads the graph data into physical RAM.  Makes a "best effort" (see MappedByteBuffer.load()).
```

Comment:
Correct, the semantics don't change when this is called.  It just does the copy up-front, to avoid latency later.  preLoad sounds good, and I'll update the comment to explain the semantics.